### ISSUE TITLE
Can't access 'group' column of electrodes table

### ISSUE THREAD

**Issue by VBaratham** [NeurodataWithoutBorders/pynwb#739]
After writing a file containing a dataset with an 'electrodes' attribute, then reading the file back from disk, an error is thrown by calling `electrodes.table.to_dataframe()`:

```
Traceback (most recent call last):
  File "tests/electrodes_mwe.py", line 40, in <module>
    electrodes = nwb.acquisition['Raw']['testdevice'].electrodes.table.to_dataframe()
  File "/home/vyassa/src/pynwb/src/pynwb/core.py", line 1212, in to_dataframe
    data[name] = col[:]
  File "/home/vyassa/src/pynwb/src/pynwb/core.py", line 294, in __getitem__
    return self.data[args]
  File "/home/vyassa/src/pynwb/src/pynwb/form/backends/hdf5/h5_utils.py", line 104, in __getitem__
    return self.io.get_container(self.dataset.file[ref])
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/home/vyassa/anaconda2/lib/python2.7/site-packages/h5py/_hl/group.py", line 167, in __getitem__
    oid = h5o.open(self.id, self._e(name), lapl=self._lapl)
  File "/home/vyassa/anaconda2/lib/python2.7/site-packages/h5py/_hl/base.py", line 132, in _e
    name = name.encode('ascii')
AttributeError: 'numpy.ndarray' object has no attribute 'encode'
```

Similar to #737 but probably a different issue

### Steps to Reproduce
```
from pynwb import NWBFile, NWBHDF5IO
from pynwb.ecephys import ElectricalSeries, LFP
from datetime import datetime
import numpy as np


nwb = NWBFile(
    session_description='foo',
    identifier='Simulation_v0',
    session_start_time=datetime.now(),
    file_create_date=datetime.now(),
    experimenter='asdf',
    experiment_description='adsf',
    session_id='Simulation_v0',
    institution='wqre',
    lab='sdfpo',
    pharmacology='dskoe',
    notes='kdoe',
    surgery='dkoie'
)

device = nwb.create_device(name='testdevice')
e_group = nwb.create_electrode_group(name='devicename', description='', location='', device=device)
electrode = nwb.add_electrode(0, x=.0, y=0.0, z=0.0, imp=1.0, location='0', filtering='Low pass', group=e_group)
electrode = nwb.add_electrode(1, x=.0, y=0.0, z=1.0, imp=1.0, location='1', filtering='Low pass', group=e_group)
e_region = nwb.create_electrode_table_region(region=[0], description='')

raw_data = np.ones(shape=(100, 1))
raw_e_series = ElectricalSeries(name='testdevice', data=raw_data, electrodes=e_region, starting_time=0.0, rate=1000.0)
nwb.add_acquisition(LFP(name='Raw'))
nwb.acquisition['Raw'].add_electrical_series(raw_e_series)

io = NWBHDF5IO('electrodes_mwe.nwb', 'w')
io.write(nwb)
io.close()

io = NWBHDF5IO('electrodes_mwe.nwb', 'a')
nwb = io.read()

electrodes = nwb.acquisition['Raw']['testdevice'].electrodes.table.to_dataframe()
```
### Environment
Please describe your environment according to the following bullet points.
- **Python Executable:** Conda
- **Python Version:** Python 3.5.1
- **Operating System:** Linux
- **Pynwb Version:** 35a52c64d6d758fe0275254ed3f49c7e43783987 (latest commit on dev branch as of 11/18)

## Checklist

- [x] Have you ensured the feature or change was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues/) ?
- [x] Have you included a brief and descriptive title?
- [x] Have you included a clear description of the problem you are trying to solve?
- [x] Have you included a minimal code snippet that reproduces the issue you are encountering?

**Comment by bendichter** [comment]
@VBaratham Thanks for the bug reports! I'll look into this right away.

edit: reproduced on osx

**Comment by VBaratham** [comment]
Thanks @bendichter ! 

Not sure if you've discovered this already, but I've done a bit more digging and found that the error is caused by trying to access the 'group' column. If you change the last few lines of the MWE above to:

```
table = nwb.acquisition['Raw']['testdevice'].electrodes.table
for i in range(len(table.colnames)):
    print(i)
    print(table.colnames[i])
    print()
    electrodes = table.columns[i][:]
```

I think you'll see what I mean

**Comment by bendichter** [comment]
@VBaratham yes, that is helpful, thank you.

Interestingly, there is no problem with accessing a single element:

```python
table = nwb.acquisition['Raw']['testdevice'].electrodes.table
for i in range(len(table.colnames)):
    print(i)
    print(table.colnames[i])
    print(table.columns[i][0])
```
runs as expected. The error occurs when reading multiple group elements at once.

**Comment by bendichter** [comment]
@VBaratham fixed it in #740. We'll merge it into dev asap and cut a new release.

**Comment by VBaratham** [comment]
Awesome, thanks for the quick fix!!
